hack/record-gcp

```diff
@@ -34,5 +40,5 @@ GOLDEN_REQUEST_CHECKS=1 \
 GOLDEN_OBJECT_CHECKS=1 \
 WRITE_GOLDEN_OUTPUT=1 \
 RUN_E2E=1 \
-KCC_USE_DIRECT_RECONCILERS="SQLInstance,ComputeForwardingRule" \
+KCC_USE_DIRECT_RECONCILERS=$KCC_USE_DIRECT_RECONCILERS \
```

Comment:
I think we can remove the `KCC_USE_DIRECT_RECONCILERS` from the `hack/record-gcp` now. @jasonvigil Could you verify the SQLInstance does not need this to run against the real gcp?

Reply:
Yeah let's just remove it. We can now use the `alpha.cnrm.cloud.google.com/reconciler: "direct"` annotation to toggle direct reconciler usage for individual mockgcp test cases. [example](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/f3e33671f78a6fa3b6540bef3e1f8aef44693d51/pkg/test/resourcefixture/testdata/basic/dataflow/v1beta1/dataflowflextemplatejob/batchdataflowflextemplatejob-direct/create.yaml#L20C5-L20C53)

Reply:
Let me remove `SQLInstance` here